### ISSUE TITLE
D3 3 broken with babel application

### ISSUE THREAD

**Issue by fheidrich** [d3/d3#2985]
Hello
I have a requireJS+babel application which uses "transform-es2015-modules-amd" plugin.
Recently I updated to d3.4.2.4 and now my transpiled code no longer works with d3.

The reason seems to be the following line at the end of  https://d3js.org/d3.v4.js
Object.defineProperty(exports, '__esModule', { value: true });

For me, during runtime everything is amd modules, and transform-es2015-modules-amd uses the same __esModule to mark the modules it has transpiled, modules without the __esModule=true are handle a bit differently.

Anyhow, d3.js apparently was build by a different build system (rollup, webpack, I don't know) which also sets __esModule=true but does not play along with babel.

I think d3 could be changed to work with babel, the first lines in d3.js it selects the module system:
(a)  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
(b) typeof define === 'function' && define.amd ? define(['exports'], factory) :
(c)  (factory((global.d3 = global.d3 || {})));

I think the "Object.defineProperty(exports, '__esModule', { value: true });" is only needed for the (a) case. The (b) case is AMD runtime, which is my case, if __esModule is not set then my code should just work. The (c) is putting d3 in the global name space, certainly that doesn't need __esModule.

In short, at the end of d3.js I wish we could have:
if (typeof exports === 'object' && typeof module !== 'undefined') {
Object.defineProperty(exports, '__esModule', { value: true });
}

Thank you
Felipe

**Comment by mbostock** [comment]
This behavior is entirely controlled by Rollup. If you’d like to change Rollup’s output for its UMD format, please file an issue over on [rollup/rollup](https://github.com/rollup/rollup/issues).

**Comment by fheidrich** [comment]
Okay, filed https://github.com/rollup/rollup/issues/998
Thanks
